packages/services/rate-limit/src/api.ts

```diff
@@ -1,50 +1,149 @@
-import { z } from 'zod';
-import { handleTRPCError } from '@hive/service-common';
-import type { FastifyRequest } from '@hive/service-common';
-import type { inferRouterInputs, inferRouterOutputs } from '@trpc/server';
-import { initTRPC } from '@trpc/server';
+import {
+  createRouter,
+  FromSchema,
+  Response,
+  RouterInput,
+  RouterOutput,
+  useErrorHandling,
+} from 'fets';
+import { createLogger } from '@graphql-yoga/logger';
+import { reportReadiness } from '@hive/service-common';
+import * as Sentry from '@sentry/node';
 import type { Limiter } from './limiter';
 
 export interface Context {
-  req: FastifyRequest;
   limiter: Limiter;
 }
 
-const t = initTRPC.context<Context>().create();
-const errorMiddleware = t.middleware(handleTRPCError);
-const procedure = t.procedure.use(errorMiddleware);
+const VALIDATION = {
```

Comment:
can we have some type checks on this object? like `RateLimitValidation: JsonSchema`?

Reply:
No because it needs to be a `const` type to infer types from JSON Schema correctly.

Reply:
Changed to Zod 👍